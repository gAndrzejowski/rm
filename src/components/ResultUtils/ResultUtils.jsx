// @flow
import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { searchAndSort } from '../../actions/async';
import styles from './ResultUtils.scss';
import type { MovieData, Store } from '../../flowTypes';

type Props = {
  numFound: number,
  currentSort: string,
  search: string,
  by: string,
  changeSorting: (string, string, string) => Promise<any>,
  selected: MovieData,
}

export const ResultUtils = ({
  numFound, currentSort, changeSorting, search, by, selected,
} :Props) => (
  <div className={styles.resultUtils}>
    {numFound > 0 && (
    <Fragment>
      <p>{(selected && selected.genres) ? `Films by ${selected.genres[0]} genre` : `${numFound} movies found`}</p>
      <div className={styles.sort}>
        <div>Sort By</div>
        <div onClick={() => changeSorting('release_date', search, by)} className={currentSort === 'release_date' ? styles.active : ''} role="button">release date</div>
        <div onClick={() => changeSorting('vote_average', search, by)} className={currentSort === 'vote_average' ? styles.active : ''} role="button">rating</div>
      </div>
    </Fragment>
            )}
  </div>
);

export const mapStateToProps = (state :Store) => ({
  numFound: state.movies.results.length,
  currentSort: state.movies.sort,
  search: state.heading.search,
  by: state.heading.by,
  selected: state.heading.selected,
});
export const mapDispatchToProps = (dispatch :any) => ({
  changeSorting: async (crit :string, search :string, by :string) =>
    dispatch(await searchAndSort(crit, search, by)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ResultUtils);
