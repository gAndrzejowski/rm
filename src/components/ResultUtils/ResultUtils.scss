.resultUtils {
  background:wheat;
  display:flex;
  justify-content: space-between;
  padding:0.5em 5vw;
  p, div {
    display:flex;
    justify-content: space-around;
    align-items:center;
    color:black;
    font-weight: bold;
  }
  .sort {
    width:20vw;
  }
}
.active {
  color:firebrick!important;
}
