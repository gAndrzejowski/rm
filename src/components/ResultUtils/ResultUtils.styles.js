const styles = {
  resultUtils: {
    background: 'wheat',
    display: 'flex',
    justifyContent: 'space-between',
    padding: ['0.5em', '5vw'],
    '@global p, div': {
      display: 'flex',
      justifyContent: 'space-around',
      alignItems: 'center',
      color: 'black',
      fontWeight: 'bold',
    },
  },
  sort: {
    width: '20vw',
    '@global div': {
      cursor: 'pointer',
    },
  },
  active: {
    color: ['firebrick', '!important'],
  },
};
export default styles;

