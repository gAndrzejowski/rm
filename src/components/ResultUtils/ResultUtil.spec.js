import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { ResultUtils } from './ResultUtils';
import styles from 'ResultUtils.scss';

describe('ResultUtil components', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<ResultUtils changeSorting={jest.fn()} currentSort="release_date" numFound={10} />);
        expect(tree).toMatchSnapshot();
    });
    it('assigns active class to proper option div based on currentSort prop', () => {
        const wrapper = shallow(<ResultUtils changeSorting={jest.fn()} currentSort="release_date" numFound={10} />)
        expect(wrapper.find(`.${styles.active}`).text()).toBe('release date');

        wrapper.setProps({currentSort: 'vote_average'});
        expect(wrapper.find(`.${styles.active}`).text()).toBe('rating');
    });
    it('calls changeSorting with proper args when one of the option divs is clicked', () => {
        const wrapper = shallow(<div><ResultUtils changeSorting={jest.fn()} currentSort="release_date" numFound={10} /></div>);
        const utils = wrapper.find('ResultUtils').first().dive();
        utils.find('div[onClick]').first().simulate('click');
        utils.find('div[onClick]').at(1).simulate('click');
        const sortMock = wrapper.find('ResultUtils').first().prop('changeSorting');
         expect(sortMock.mock.calls.length).toBe(2);
        expect(sortMock.mock.calls[0][0]).toBe('release_date');
        expect(sortMock.mock.calls[1][0]).toBe('vote_average');
    });
});
