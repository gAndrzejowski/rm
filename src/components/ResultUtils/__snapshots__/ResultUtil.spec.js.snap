// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultUtil components renders correctly 1`] = `
<div
  className="resultUtils"
>
  <p>
    10 movies found
  </p>
  <div
    className="sort"
  >
    <div>
      Sort By
    </div>
    <div
      className="active"
      onClick={[Function]}
    >
      release date
    </div>
    <div
      className=""
      onClick={[Function]}
    >
      rating
    </div>
  </div>
</div>
`;
